add CanSlopeSpark to logic

nn357 opened the pull request:
add canSlopeSpark to logic + 2 examples in spore spawn farming room

blkerby commented:
- Both strats need a shinecharge requirement: https://github.com/vg-json-data/sm-json-data/blob/master/logicalRequirements.md#canshinecharge-object . If the usable runway is at least 45 tiles, you can use the `h_shinechargeMaxRunway` helper; otherwise you'll want to describe the usable runway for the shinecharge.
- Both strats need a `shinespark` requirement: https://github.com/vg-json-data/sm-json-data/blob/master/logicalRequirements.md#shinespark.
- The `canSlopeSpark` tech needs `canCarryFlashSuit` in its `techRequires`. It should also be an extension of `canCarryFlashSuit`, placing it in `tech.json` next to `canSpikeSuit` and `canElevatorCrystalFlash`.
- Some of the open curly braces `{` are not indented properly. We have an auto-formatting script that would eventually fix these, but it's more ideal if it is already right.

kjbranch commented on region/brinstar/pink/Spore Spawn Farming Room.json:
```diff
@@ -369,6 +369,27 @@
+      "link": [2, 2],
+      "name": "Gain Flash Suit (Slopespark) - Leave With Shinespark",
+      "requires": [
+        "canSlopeSpark"
```
I wonder if leaving through the door with a spark while gaining a flash suit should have another tech added onto it? For now, I would maybe just throw on `canComplexCarryFlashSuit`

nn357 replied:
thank you for the feedback, I stuck a cancomplexcarryflashsuit on to it.

kjbranch reviewed (commented):
Sounds like youre on a good track and are planning on making some changes soon. I'll review it again tomorrow

nn357 commented:
Thank you both for the pointers / suggestions... I added the requirements I think are necessary along with the complex carry flashsuit suggested by Kyle for the (leave with spark). Moved the tech into the suggested position in the tech.json. I added the shineframe requirements too and the enemy contact hits.

While the are technically 22/21 tiles available I subtracted 1 tile because if you stop on the tile next to the pipe you take a hit from the zeb so that tile seems like it shouldn't be included? I hope that is the correct approach?

nn357 commented on region/brinstar/pink/Spore Spawn Farming Room.json:
```diff
@@ -369,6 +369,53 @@
+          ]}
+        ]},
+        {"enemyDamage": {"enemy": "Zeb", "type": "contact", "hits": 1}}
+      ],
```
{"shinespark": {"frames": 4, "excessFrames": 4}} so like this? (it takes 4 energy if you hit the ceiling with a diagonal spark) but you can still get a flash suit with less than 29 energy just your spark stops straight away.

blkerby commented on region/brinstar/pink/Spore Spawn Farming Room.json:
```diff
@@ -369,6 +369,58 @@
+            {"canShineCharge": {"usedTiles": 21, "steepDownTiles": 2, "openEnd": 0}}
+          ]}
+        ]},
+        {"shineChargeFrames": 143},
```
```suggestion
        {"shineChargeFrames": 150},
```
We usually round up to multiples of 5 for frame counts. And it's better if the value in logic isn't tighter than what the video demonstrates. For a strat like this where the shinecharge is gained and used in the same room, it's not necessary to include a `shineChargeFrames` requirement at all. Personally I probably wouldn't bother with it except in a case where it was particularly tight (10 or fewer frames remaining). But it's also ok here as a way of describing how much extra time you have.

blkerby commented on region/brinstar/pink/Spore Spawn Farming Room.json:
```diff
@@ -369,6 +369,58 @@
+        {"shinespark": {"frames": 4, "excessFrames": 4}}
+      ],
+      "note": [
+        "After shinecharging, lure a zeb and use it to slope spark on first slope right of the pipe"
```
```suggestion
        "After shinecharging, lure a Zeb and use it to slopespark on the first slope right of the pipe."
```

blkerby commented on region/brinstar/pink/Spore Spawn Farming Room.json:
```diff
@@ -369,6 +369,58 @@
+        }
+      },
+      "note": [
+        "After shinecharging, lure a zeb and use it to slope spark on first slope right of the pipe"
```
```suggestion
        "After shinecharging, lure a Zeb and use it to slopespark on the first slope right of the pipe."
```

blkerby commented on tech.json:
```diff
@@ -2290,6 +2290,25 @@
+              "detailNote": [
+                "A common failure is switching directions too late. The frame perfect direction switch is very early, one frame before the damage pose appears.",
+                "This can be difficult to judge, especially with faster moving enemies. Switching directions on the correct frame will give a very small damage boost, turning around and landing on the slope.",
+                "Holding back and jump for more than 5 frames will cause samus to spark horizontally, if the direction switch was exactly one frame late Samus will land on the slope but no spark will occur." 
```
```suggestion
                "Holding back and jump for more than 5 frames will cause Samus to spark horizontally. If the direction switch was exactly one frame late, Samus will land on the slope but no spark will occur." 
```

kjbranch commented on region/brinstar/pink/Spore Spawn Farming Room.json:
```diff
@@ -369,6 +369,58 @@
+      "name": "Gain Flash Suit (Slopespark) - Leave With Shinespark",
+      "requires": [ 
+        "canComplexCarryFlashSuit",
+        {"or": [
+          {"canShineCharge": {"usedTiles": 20, "steepDownTiles": 2, "openEnd": 0}},
+          {"and": [
+            {"doorUnlockedAtNode": 2},
+            {"canShineCharge": {"usedTiles": 21, "steepDownTiles": 2, "openEnd": 0}}
+          ]}
+        ]},
```
The door will have to be open for this strat
```suggestion
          {"canShineCharge": {"usedTiles": 21, "steepDownTiles": 2, "openEnd": 0}},
```
